BreakingBad/CharacterListScreen.swift

```diff
@@ -56,46 +56,56 @@ class CharacterListScreen: UIViewController {
     
      
     private func createCharacterList(from characterList : [CharacterJson]) -> [Character] {
-
         var tempCharacters: [Character] = []
-
         for character in characterList {
-            //let imageURL = URL(string: character.img)
+            let image = defaultImage!
             let birthday: String = character.birthday
             let age = addAgeToBirthday(from: birthday)
-            let image = defaultImage! //getImage(from: imageURL)
             let newCharacter = Character(name: character.name, birthday: age, image: image, nickname: character.nickname, occupations: character.occupation, portrayed: character.portrayed)
             tempCharacters.append(newCharacter)
-            
+            if let imageURL = URL(string: character.img) {
+                getImageAsync(from: newCharacter, imageUrl: imageURL)
+            }
         }
         return tempCharacters
     }
     
     private func addAgeToBirthday(from birthday: String) -> String {
```

Comment:
There are two methods hidden in this one method. 1) getting a date from the birthday; 2) getting the age text in brackets from the date. Try break this method up into smaller components. You should be left with three methods in total when you are done.